presto-main/src/main/java/com/facebook/presto/operator/OperationTimer.java

```diff
@@ -0,0 +1,285 @@
+            }
+            else {
+                overallTiming.recordEstimatedCpuTime(wallDuration);
+                overallTiming.recordEstimatedUserTime(wallDuration);
+            }
+        }
+
+        long totalUnaccountedCpuWallNanos = timings.values().stream()
+                .mapToLong(InternalTiming::getCpuUnaccountedWallNanos)
+                .sum();
+
+        long totalMeasuredCpuNanos = timings.values().stream()
+                .mapToLong(InternalTiming::getCpuMeasuredNanos)
+                .sum();
+        long totalUnaccountedCpuNanos = max(0, cpuEnd - cpuStart - totalMeasuredCpuNanos);
+
+        long totalMeasuredUserNanos = timings.values().stream()
+                .mapToLong(InternalTiming::getUserMeasuredNanos)
+                .sum();
+        long totalUnaccountedUserNanos = max(0, userEnd - userStart - totalMeasuredUserNanos);
+
+        for (InternalTiming timing : timings.values()) {
+            timing.recordTimings(totalUnaccountedCpuNanos, totalUnaccountedUserNanos, totalUnaccountedCpuWallNanos);
+        }
+    }
+
+    private static long currentThreadUserTime()
+    {
+        return THREAD_MX_BEAN.getCurrentThreadUserTime();
+    }
+
+    private static long currentThreadCpuTime()
+    {
+        return THREAD_MX_BEAN.getCurrentThreadCpuTime();
+    }
+
+    private static class InternalTiming
+    {
+        private final OperationTiming operationTiming;
+
+        private long cpuMeasuredNanos;
+        private long userMeasuredNanos;
+        private long cpuUnaccountedWallNanos;
+
+        public InternalTiming(OperationTiming operationTiming)
+        {
+            this.operationTiming = operationTiming;
+        }
+
+        public void recordMeasuredCpuTime(long intervalWallNanos, long intervalCpuNanos, long intervalUserNanos)
+        {
+            operationTiming.recordWallTime(intervalWallNanos);
+
+            cpuMeasuredNanos += intervalCpuNanos;
+            userMeasuredNanos += intervalUserNanos;
+        }
+
+        public void recordUnaccountedCpuTime(long intervalWallNanos)
+        {
+            operationTiming.recordWallTime(intervalWallNanos);
+            cpuUnaccountedWallNanos += intervalWallNanos;
+        }
+
+        public long getCpuMeasuredNanos()
+        {
+            return cpuMeasuredNanos;
+        }
+
+        public long getCpuUnaccountedWallNanos()
+        {
+            return cpuUnaccountedWallNanos;
+        }
+
+        public long getUserMeasuredNanos()
+        {
+            return userMeasuredNanos;
+        }
+
+        public void recordTimings(long totalUnaccountedCpuNanos, long totalUnaccountedUserNanos, long totalUnaccountedCpuWallNanos)
+        {
+            if (cpuUnaccountedWallNanos > 0) {
+                // measured CPU + this timers portion of the unaccounted CPU
+                operationTiming.recordEstimatedCpuTime(cpuMeasuredNanos + totalUnaccountedCpuNanos * (totalUnaccountedCpuWallNanos / cpuUnaccountedWallNanos));
```

Comment:
It seems that the formula should be next `totalUnaccountedCpuNanos * currentOperatorUnaccountedFraction`, where `currentOperatorUnaccountedFraction = cpuUnaccountedWallNanos / (double) totalUnaccountedCpuWallNanos`